controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -1682,9 +1682,15 @@ func (r *BareMetalHostReconciler) updateEventHandler(e event.UpdateEvent) bool {
 }
 
 // SetupWithManager registers the reconciler to be run by the manager
-func (r *BareMetalHostReconciler) SetupWithManager(mgr ctrl.Manager, preprovImgEnable bool) error {
+func (r *BareMetalHostReconciler) SetupWithManager(mgr ctrl.Manager, preprovImgEnable bool, options controller.Options) error {
+
+	maxConcurrentReconciles := 0
+	if options.MaxConcurrentReconciles > 0 {
+		maxConcurrentReconciles = options.MaxConcurrentReconciles
+	} else {
+		maxConcurrentReconciles = runtime.NumCPU()
+	}
```

Comment:
Do we want to replicate this logic and the handling of BMO_CONCURRENCY below in other controllers?

Reply:
Not sure since the other controllers don’t have it currently, I didn’t add it. Wdyt ?